Imagine/Filter/PostProcessor/JpegOptimPostProcessor.php

```diff
@@ -46,129 +40,135 @@ class JpegOptimPostProcessor implements PostProcessorInterface
-    public function setStripAll($stripAll)
+    public function setStripAll($strip)
     {
-        $this->stripAll = $stripAll;
+        $this->triggerSetterMethodDeprecation(__METHOD__);
+        $this->strip = $strip;
 
         return $this;
     }
 
     /**
-     * @param BinaryInterface $binary
-     * @param array           $options
-     *
-     * @throws ProcessFailedException
-     *
-     * @return BinaryInterface
+     * @inheritDoc
      */
     public function process(BinaryInterface $binary, array $options = []): BinaryInterface
     {
-        $type = mb_strtolower($binary->getMimeType());
-        if (!\in_array($type, ['image/jpeg', 'image/jpg'], true)) {
+        if (!$this->isBinaryTypeJpgImage($binary)) {
             return $binary;
         }
 
-        $tempDir = \array_key_exists('temp_dir', $options) ? $options['temp_dir'] : $this->tempDir;
-        if (false === $input = tempnam($tempDir, 'imagine_jpegoptim')) {
-            throw new \RuntimeException(sprintf('Temp file can not be created in "%s".', $tempDir));
-        }
+        $file = $this->writeTemporaryFile($binary, $options, 'imagine-post-processor-jpegoptim');
 
-        $processArguments = [$this->jpegoptimBin];
+        $arguments = $this->getProcessArguments($options);
+        $arguments[] = $file;
+        $process = $this->createProcess($arguments, $options);
+        $process->run();
 
-        $stripAll = \array_key_exists('strip_all', $options) ? $options['strip_all'] : $this->stripAll;
-        if ($stripAll) {
-            $processArguments[] = '--strip-all';
+        if (!$this->isSuccessfulProcess($process)) {
+            unlink($file);
+            throw new ProcessFailedException($process);
         }
 
-        $max = \array_key_exists('max', $options) ? $options['max'] : $this->max;
-        if ($max) {
-            $processArguments[] = '--max='.$max;
-        }
+        $result = new Binary(file_get_contents($file), $binary->getMimeType(), $binary->getFormat());
 
-        $progressive = \array_key_exists('progressive', $options) ? $options['progressive'] : $this->progressive;
-        if ($progressive) {
-            $processArguments[] = '--all-progressive';
-        } else {
-            $processArguments[] = '--all-normal';
-        }
+        unlink($file);
 
-        $processArguments[] = $input;
-        if ($binary instanceof FileBinaryInterface) {
-            copy($binary->getPath(), $input);
-        } else {
-            file_put_contents($input, $binary->getContent());
+        return $result;
+    }
+
+    /**
+     * @param array $options
+     *
+     * @return array
+     */
+    private function getProcessArguments(array $options = []): array
+    {
+        $arguments = [$this->executablePath];
+
+        if ($options['strip_all'] ?? $this->strip) {
+            $arguments[] = '--strip-all';
         }
 
-        $proc = new Process($processArguments);
-        $proc->run();
+        if (isset($options['max'])) {
```

Comment:
could we do this in the dependency injection class already, and maybe also normalize the config, instead of at runtime?